File: backend/voting-app/src/main/java/com/tu/votingapp/utils/mappers/DocumentMapper.java
package com.tu.votingapp.utils.mappers;

import com.tu.votingapp.dto.general.DocumentDTO;
import com.tu.votingapp.entities.DocumentEntity;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(componentModel = "spring", uses = {UserMapper.class})
public interface DocumentMapper {
    @Mapping(target = "user", ignore = true)
    DocumentDTO toDto(DocumentEntity entity);
    DocumentEntity toEntity(DocumentDTO dto);
}

File: frontend/android/app/src/main/java/com/example/android/utils/mappers/DocumentMapper.kt
package com.example.android.utils.mappers

import com.example.android.dto.response.DocumentDTO
import com.example.android.dto.response.UserDTO
import com.example.android.entity.DocumentEntity

object DocumentMapper {

    fun DocumentDTO.toEntity(userId: Long): DocumentEntity = DocumentEntity(
        id = this.id?:1,
        number = this.number,
        validFrom = this.validFrom,
        validTo = this.validTo,
        issuer = this.issuer,
        gender = this.gender,
        dateOfBirth = this.dateOfBirth,
        permanentAddress = this.permanentAddress,
        userId = userId
    )

    fun DocumentEntity.toDto(): DocumentDTO = DocumentDTO(
        id = this.id,
        number = this.number,
        validFrom = this.validFrom,
        validTo = this.validTo,
        issuer = this.issuer,
        gender = this.gender,
        dateOfBirth = this.dateOfBirth,
        permanentAddress = this.permanentAddress,
    )
}